alternatives.go

```diff
@@ -0,0 +1,105 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+)
+
+// iptablesAlternativeSelector allows to configure a system to use iptables in
+// nft or legacy mode.
+type iptablesAlternativeSelector interface {
+	// useMode configures the system to use the selected iptables mode.
+	useMode(ctx context.Context, mode iptablesMode) error
+}
```

Comment:
This feels like more abstraction than we need (especially given that one of the goals is to be simple and CVE-free). You can just combine `buildIptablesAlternativeSelector` and `useMode` into a single function.

Reply:
Just curious, when you mention CVE-free, what's your concern with this specific piece of code? Like using an interface could extend the attack surface?

Reply:
no, just "more code, more chances for bugs"

Reply:
Got you. That makes sense, although I'm not sure this specific instance opens the door for bugs.

However, continuing in that line of thought, what if we just remove the `alternatives` and `update-alternatives` logic? I believe the "default" logic that handles symlinks manually should work whether `alternatives`/`update-alternatives` are present or not.

That would have a significant impact in the amount of code and, in fact, make this interface redundant.

Reply:
It _would_ be simpler, but it would be fragile too. For example, if you have a base image containing `iptables-wrapper` with "manual" symlinks, and then someone builds another image on top of that and runs `apt-get update` as part of their build, and gets a newer version of iptables in the process, then that would overwrite your symlinks and revert them back to what `update-alternatives` thought they were supposed to be.

Reply:
Oh I didn't think about that usecase, thank you! However there is something else I'm missing I think.

If someone tried to build an image using an iptables-wrapper image as a base, the symlinks present in the image are still pointing to the wrapper (since it has never been executed yet). And those symlinks were created (most probably) with the installer script, which it does use `update-alternatives`/`alternatives`. So running `apt-get update` on top of that should respect the selected "alternative".

So, does it really affect this usecase if the wrapper itself creates manual symlinks? Those symlinks would be created in runtime, once the first `iptables` command is run and always after all build processes have finished (no matter how many images on top of images the user builds).